### PR TITLE
[metricbeat] fix aws cloudwatch metric tags when the number of resource is a multiple of 50 (#26385)

### LINKED ISSUE #26385: [MetricBeat] [AWS] if the number of resources is a multiple of 50, there is no tags field in the event
## the problem
I found a strange problem of AWS cloudwatch module,  there are some EC2 instances cross some regions, but sometimes metricbeat event of some region has no tags.   I can query tags with aws CLI `aws resourcegroupstaggingapi`, then I think there is some bug in the logic of querying tags.

- Version: 7.13
- Operating System: linux
- Steps to Reproduce: 

    * Some AWS resource's amount is a multiple of 50(e.g. 100 EC2 instances)
    * Use aws CloudWatch Module 

metricbeat config: 

```yaml

  - module: aws
    access_key_id: 'your access key'
    secret_access_key: 'your secret key'
    period: 300s
    regions:
      - ap-northeast-1
    metricsets:
      - cloudwatch
    metrics:
      - namespace: AWS/EC2
        name: CPUUtilization
        statistic: ["Average"]
        resource_type: "ec2:instance"
```

## the reason 

* aws resourcegroupstaggingapi has strange behavior, with the default pagination size of 50 and 100 EC2 instances, we need send three times request and there is zero result in the last request
* line 191 is not safe and correct, we should break the for loop if `output.ResourceTagMappingList` is an empty result, then the function can return `resourceTagMap` to the caller.

https://github.com/elastic/beats/blob/6347d455ead8b9af6bf8a49eaa9615f0e44d21e2/x-pack/metricbeat/module/aws/utils.go#L169-L195

I modify the `if` statemen, then metricebeat event has tags field.

```go
if len(output.ResourceTagMappingList) == 0 {  	
    break
} 
```

### PR DESCRIPTION
**Pull request by kwinstonix**
## What does this PR do?

see issue #26385

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes #26385

### CODE REVIEW

File: x-pack/metricbeat/module/aws/utils.go [hunk 1]
```diff
@@ -188,8 +188,8 @@ func GetResourcesTags(svc resourcegroupstaggingapiiface.ClientAPI, resourceTypeF
 		}
 
 		getResourcesInput.PaginationToken = output.PaginationToken
-		if resourceTypeFilters == nil || len(output.ResourceTagMappingList) == 0 {
-			return nil, nil
+		if len(output.ResourceTagMappingList) == 0 {
```
Comment (kaiyan-sheng) [inline x-pack/metricbeat/module/aws/utils.go:191]: Thank you for fixing this! I almost created a PR to fix pagination in aws module yesterday but you beat me to it!! One question, what do you think about using the paginator in aws sdk? For example here: https://github.com/aws/aws-sdk-go-v2/blob/main/service/resourcegroupstaggingapi/api_op_GetResources.go#L261

Comment (kwinstonix) [reply x-pack/metricbeat/module/aws/utils.go:191]: yes，SDK paginator is easy and safe.

Comment (kwinstonix) [reply x-pack/metricbeat/module/aws/utils.go]: It is done.

**Comment by kwinstonix** [comment]
rebased on upstream/master

**Comment by kwinstonix** [comment]
failed test cases are fixed. please run CI test again.  😅

**Comment by kaiyan-sheng** [comment]
@kwinstonix Thank you!!

**Review by kaiyan-sheng — APPROVED** [review]

**Pull request merged**